tendrl/commons/flows/unmanage_cluster/__init__.py

```diff
@@ -14,59 +21,203 @@ def run(self):
+                )
+                etcd_keys_to_delete.append(
+                    "/clusters/%s/Utilization" % integration_id
+                )
+                etcd_keys_to_delete.append(
+                    "/clusters/%s/raw_map" % integration_id
+                )
+                etcd_keys_to_delete.append(
+                    "/alerting/clusters/%s" % integration_id
+                )
+                nodes = etcd_utils.read(
+                    "/clusters/%s/nodes" % integration_id
+                )
+                node_ids = []
+                for node in nodes.leaves:
+                    node_id = node.key.split("/")[-1]
+                    node_ids.append(node_id)
+                    key = "/alerting/nodes/%s" % node_id
+                    etcd_keys_to_delete.append(
+                        key
+                    )
+                    try:
+                        # delete node alerts from /alerting/alerts
+                        node_alerts = etcd_utils.read(key)
+                        for node_alert in node_alerts.leaves:
+                            etcd_keys_to_delete.append(
+                                "/alerting/alerts/%s" % node_alert.key.split(
+                                    "/")[-1]
+                            )
+                    except etcd.EtcdKeyNotFound:
+                        # No node alerts, continue
+                        pass
+
+                # Find the alerting/alerts entries to be deleted
+                try:
+                    cluster_alert_ids = etcd_utils.read(
+                        "/alerting/clusters/%s" % integration_id
+                    )
+                    for entry in cluster_alert_ids.leaves:
+                        ca_id = entry.key.split("/")[-1]
+                        etcd_keys_to_delete.append(
+                            "/alerting/alerts/%s" % ca_id
+                        )
+                except etcd.EtcdKeyNotFound:
+                    # No cluster alerts, continue
+                    pass
+
+                # Remove the cluster details
+                for key in list(set(etcd_keys_to_delete)):
+                    try:
+                        etcd_utils.delete(key, recursive=True)
+                    except etcd.EtcdKeyNotFound:
+                        logger.log(
+                            "debug",
+                            NS.publisher_id,
+                            {
+                                "message": "%s key not found for deletion" %
+                                           key
+                            },
+                            job_id=self.parameters['job_id'],
+                            flow_id=self.parameters['flow_id'],
+                        )
+                        continue
+                # remove short name
+                _cluster = NS.tendrl.objects.Cluster(
+                    integration_id=integration_id
+                ).load()
+                _cluster.short_name = ""
```

Comment:
Why short_name to be un-set after un-manage? After un-manage short_name should be retained I feel, and while re-import user should have option to provide a different name or retain the same.
@nthomas-redhat @r0h4n @julienlim thoughts?

Reply:
So what I feel is, if cluster.short_name available while import populate the same in UI screen where list of nodes to be imported are listed. User still should be able to edit the field.

Reply:
@shtripat @anmolsachan +1 that it would be nice to retain the short_name during the unmanage if possible so that if the same cluster was imported again, user had the option to reuse the same short name.  However, if it's too complex, then we defer to this in the future as I think it's a nice-to-have.

@nthomas-redhat

Reply:
@gnehapk Will this be a big change for UI  ? Can this be done ?

Reply:
@anmolsachan Its a small change in UI. UI will by populate cluster name field with short_name if exist else integration_id on import cluster flow. The cluster name is a editable field and can be changed by user. Created issue for the same https://github.com/Tendrl/ui/issues/973

Reply:
@gnehapk @shtripat Back-end will save the cluster name now. Made the required change